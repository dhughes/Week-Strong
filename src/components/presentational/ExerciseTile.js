import React from 'react';
import styled from 'styled-components';
import { Vbox, Hbox } from './Box';
import { Checkmark, Close } from './Icon';
import theme from '../../util/theme';

const Exercise = styled(Vbox)`
  border: 1px solid ${theme.primaryText};
  box-sizing: border-box;
  width: 44vw;
  height: 44vw;
  margin-bottom: 1em;
  background-size: cover;
  background-position: 50% 100%;
  border-radius: 0.3em;

  background-image: url(${props => props.image});

  & > * {
    margin: 0;
    padding: 0;
    height: 3rem;
    line-height: 3rem;
  }

  & > h3{
    background-color: ${theme.primary.fade(0.25)};
    color: ${theme.primaryText.negate()};
    align-self: flex-end;
  }

  & > div{
    height: 3rem;
    margin-bottom: -3rem;
    line-height: 3rem;
    background-color: ${theme.primaryText.negate().fade(0.25)};
    justify-content: flex-start;

    & > * {
      height: 3rem;
      width: 2rem;
      margin: 0.2rem;
      color: ${theme.positive};
    }

    & > h2{
      text-align: left;
      flex-grow: 1000;
    }

    & > svg {
      fill: currentColor;
    }

    & > .remove{
      fill: ${theme.negative};
    }
  }
`;

const ExerciseTile = props =>
  <Exercise reverse={true} justifyContent="space-between" image={props.image} onClick={props.onSelect}>
    <h3>{props.label}</h3>

    {props.goal
      ? <Hbox>
          <Checkmark />
          <h2>{props.goal}</h2>
          <Close className="remove" onClick={props.onRemove} />
        </Hbox>
      : null}
  </Exercise>;

export default ExerciseTile;
