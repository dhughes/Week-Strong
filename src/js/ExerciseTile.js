import React, { Component } from 'react';
import styled from 'styled-components';
import { Vbox, Hbox } from './Box';
import { Checkmark, Close } from './Icon';

const CenteredVbox = styled(Vbox)`
  & > * {
     margin: 0;
     width: 100%;
     text-align: center;
   }
`;

const SelectableImageHbox = styled(Hbox)`
  background-size: cover;
  background-position: 50% 100%;
  min-height: 200px;

  & > div {
    flex: 1;
    display: flex;
    justify-content: center;
    align-items: center;
    vertical-align: middle;
    font-size: 2em;
    font-weight: bold;
    color: green;
    padding: 0.25em;
    background-color: rgba(255,255,255,0.75);

    & > img {
        width: 66%;
        max-width: 150px;
        height: 66%;
    }
  }
`;

const Selected = styled(Hbox)`
  height: 3em;
  margin-bottom: -3em;
  line-height: 3em;
  background-color: rgba(238, 235, 211, 0.9);
  color: ${props => props.theme.positive};;
  z-index: 1;
  justify-content: flex-start;

  & > * {
    height: 2em;
    width: 2em;
    margin: 0.2em;
  }

  & > h2{
    text-align: left;
    flex-grow: 1000;
  }

  & > svg {
    fill: currentColor;
  }

  & > button{
    padding: 0;
    fill: ${props => props.theme.warning};;
    margin: 0;
    padding: 0;
    height: 4.5em;
    width: 4.5em;
    z-index: 2;
  }

  & > .remove {
    height: 3em;
    width: 3em;
  }
`;

const Label = styled(Hbox)`
  height: 3em;
  margin-top: -3em;
  line-height: 3em;
  background-color: rgba(57, 62, 65, 0.75);
  color: ${props => props.theme.font};;
`;

class ExerciseTile extends Component {
  onRemove = e => {
    e.stopPropagation();
    e.preventDefault();
    this.props.onRemove(e);
  };
  render() {
    return (
      <CenteredVbox onClick={this.props.onSelect}>
        {this.props.isSelected
          ? <Selected>
              <Checkmark />
              <h2>{this.props.count}</h2>
              <button title="Remove" onClick={this.onRemove}>
                <Close className="remove" />
              </button>
            </Selected>
          : null}
        <SelectableImageHbox
          className="exercise"
          style={{
            backgroundImage: `url(${this.props.src})`
          }}
          title={this.props.label}
        />
        <Label>
          <h4>
            {this.props.label}
          </h4>
        </Label>

      </CenteredVbox>
    );
  }
}

ExerciseTile.defaultProps = {
  isSelected: false,
  onSelect: () => {},
  onRemove: () => {}
};

export default ExerciseTile;
