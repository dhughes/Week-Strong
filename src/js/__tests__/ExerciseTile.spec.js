import React from 'react';
import { shallow, mount } from 'enzyme';
import ExerciseTile from '../ExerciseTile';

test('ExerciseTile renders', () => {
  const component = shallow(<ExerciseTile />);
  expect(component).toMatchSnapshot();
});

test('ExerciseTile shows "Test" label', () => {
  const component = shallow(<ExerciseTile label="Test" />);

  expect(component.find('h4').text()).toEqual('Test');
});

test('ExerciseTile shows "Pushups" label', () => {
  const component = shallow(<ExerciseTile label="Pushups" />);
  expect(component.find('h4').text()).toEqual('Pushups');
});

test('ExerciseTile shows checkmark when isSelected', () => {
  const component = shallow(<ExerciseTile label="Pushups" src="/img/pullups.jpg" isSelected={true} />);
  expect(component.find('Checkmark').length).toEqual(1);
});

test('ExerciseTile does not show checkmark when not isSelected', () => {
  const component = shallow(<ExerciseTile label="Pushups" src="/img/pullups.jpg" isSelected={false} />);
  expect(component.find('Checkmark').length).toEqual(0);
});

test('ExerciseTile checkmark defaults to not selected', () => {
  const component = shallow(<ExerciseTile label="Pushups" src="/img/pullups.jpg" />);
  expect(component.find('Checkmark').length).toEqual(0);
});

test('ExerciseTile shows pullups as title', () => {
  const component = mount(<ExerciseTile label="Pullups" src="/img/pullups.jpg" isSelected />);
  expect(component.find('Styled(styled.div)').filterWhere(n => n.prop('title') === 'Pullups').length).toBe(1);
});

test('ExerciseTile defaults to not isSelected', () => {
  const component = shallow(<ExerciseTile label="Pushups" src="/img/pullups.jpg" isSelected={false} />);
  expect(component.find('.selected').length).toEqual(0);
});

test('ExerciseTile does not show count when not isSelected', () => {
  const component = shallow(<ExerciseTile label="Pushups" src="/img/pullups.jpg" count={100} />);
  expect(component.find('.count').length).toEqual(0);
});

test('ExerciseTile shows count when isSelected', () => {
  const component = mount(<ExerciseTile label="Pushups" src="/img/pullups.jpg" isSelected count={100} />);
  expect(component.find('h2').text()).toEqual(100 + '');
});

test('ExerciseTile shows pullups image', () => {
  const component = shallow(<ExerciseTile src="/img/pullups.jpg" />);
  expect(component.html().indexOf('/img/pullups.jpg')).not.toBe(-1);
});

test('ExerciseTile shows x when isSelected', () => {
  const component = mount(<ExerciseTile label="Pushups" src="/img/pullups.jpg" isSelected count={100} />);
  expect(component.find('button').length).toEqual(1);
});
