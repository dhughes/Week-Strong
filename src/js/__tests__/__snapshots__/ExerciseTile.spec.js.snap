// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseTile renders 1`] = `
<Styled(styled.div)
  onClick={[Function]}
>
  <Styled(styled.div)
    className="exercise"
    style={
      Object {
        "backgroundImage": "url(undefined)",
      }
    }
  />
  <Styled(styled.div)>
    <h4 />
  </Styled(styled.div)>
</Styled(styled.div)>
`;
